lib/cancan/controller_resource_finder.rb

```diff
@@ -13,7 +13,9 @@ def find_resource
     end
 
     def find_resource_using_find_by
-      if resource_base.respond_to? 'find_by'
+      if resource_base.respond_to? "find_by_#{@options[:find_by]}!"
+        resource_base.send("find_by_#{@options[:find_by]}!", id_param)
+      elsif resource_base.respond_to? 'find_by'
         resource_base.send('find_by', @options[:find_by].to_sym => id_param)
```

Comment:
I know it didn't start in this PR, but why are we calling .send() on this rather than going direct in this case, such as:

resource_base.find_by(@options[:find_by].to_sym => id_param)

Reply:
Right, I think we can call `find_by` directly like that 👍  That would be great.